### PR TITLE
Tests, Line transposition, and Reaper Note Name Map exporter

### PR DESCRIPTION
**Pull request by vsicurella**
Various functions for stacking intervals represented by a "line" of any type.

Reaper Note Name Map exporter will create a text file comprised of the 128 MIDI notes with a label next to each one. This exporter has a modal that allows for several different label types and options. This text file can be loaded into Reaper for it's piano roll note display. Lots of room for more label customization.

### CODE REVIEW

File: src/js/helpers.js [hunk 1]
```diff
@@ -40,8 +45,8 @@ function commadecimal_to_decimal(input) {
 
 // convert a decimal (1.25) into commadecimal (1,25)
 function decimal_to_commadecimal(input) {
-  if (isCents(input)) { // a bit misleading
-    return input.toString().replace('.', ',');
+  if (/^\d+\.?\d*$/.test(input)) {
+    return input.toFixed(6).replace('.', ',');
```
Comment (meszaros-lajos-gyorgy) [inline src/js/helpers.js:49]: This will conflict with the other PRs

Comment (vsicurella) [reply src/js/helpers.js:49]: Should be okay if PR #110 is pulled first

**Comment by meszaros-lajos-gyorgy** [comment]
As stated in the other PR, your branch seems to be out of sync with this repo's `develop` branch. Please try pulling the latest develop branch, because there are some changes in the PR, which have already been added.

**Comment by vsicurella** [comment]
I've written a bunch of tests which greatly helped me improve the accuracy of the "LineType arithmetic" functions. They should now be solid in most cases. This includes some invalid domain handling, but I didn't cover negative values in with ratios and decimals as I'm not sure how we want to handle them. 

I also added notation for the selected options of the Reaper Note Names exporter as a suffix for the filename, so different options shouldn't require the user renaming the file. Thanks to the arithmetic tests, I'm confident in results of the exporter, except in edge-case scales and options. 
For example, make a Rank-2 scale with 16/15 as the generator, and 9/8 as the period, with the size 2. Then export Reaper Note Names with "Scale Data" and "Calculate period in pitch" enabled, and view the resulting text file. Also try the same thing without the calculated period. Most of it makes sense...but is maybe not ideal. 
I can think of some workarounds, like just converting to cents when numbers blow up. If your period is not 2/1, and you don't calculate periods in the pitch, maybe there can be an toggle option to octave reduce each period so you get a better sense of where you are in relation to true unison? It gets a little convoluted.

**Comment by vsicurella** [comment]
Oh also, I think "stackLines" and "transposeLines" might be trying to do the same thing @SeanArchibald? "transposeLines" is a much better name so if this is true then we can just rename "stackLines" and remove the other "transposeLines".

**Comment by SeanArchibald** [comment]
I didn't realise we already had stackLines. Yes if it does the same thing as transposeLines then we should keep just one of them.
Will try to test this new PR soon and get it merged in. Thanks a lot for your effort!

**Comment by vsicurella** [comment]
Cool :) I just updated the "stack" terminology to "transpose" and removed the extra function, so it should be all ready for your review! (except maybe for tweaks for those edge-cases of the Reaper exporter, but I'm interested in your feedback on that)

**Comment by SeanArchibald** [comment]
If we're changing from "stack" to "transpose", should functions like stackNOfEDOs also get renamed?

Is there anything specific that I should check with the Reaper Note Name Map? I haven't actually used it before and don't have access to reaper here, but I'll try and offer feedback if I can.

So far what I'm seeing looks good! Just checked out this branch on my local and having a play with it now.

**Comment by vsicurella** [comment]
Sensible point! I'll push an update for that shortly.

I was looking for some edge cases to share with you and I found some issues when a negative cents or N of EDO value is used as the transposer. That's a valid transposer but not a valid scale interval, so it requires a special case. I'll add tests for this, fix it up too, and report back!

**Comment by vsicurella** [comment]
I'd like to do a little more testing, but I am now handling internal negative intervals properly. 

However I did notice a parallel issue that affects the Reaper exporter - if you generate a scale with a size of M, but erase it and manually enter a scale with size N  where N < M, the tuning_table.scale_data array stays the same size and only updates the first M indexes. I'm using that array if a user wants to preserve the scale_data's line types, so the output gets messed up.  

We can address that by resetting that array each time it's updated by new input. I can handle that unless you're interested!

**Comment by vsicurella** [comment]
Never mind on that last comment, I just wasn't using the tuning_table.note_count property!

I also added descriptions for the Korg and Reaper exporter options in the guide. I thought it would be nice to show an example image for how it looks in Reaper but I can shorten that if you feel it goes beyond the scope of Scale Workshop.
